base_layer/core/src/mempool/sync_protocol/mod.rs

```diff
@@ -237,23 +364,23 @@ impl MempoolSyncProtocol {
     fn handle_protocol_notification(&mut self, notification: ProtocolNotification<Substream>) {
         match notification.event {
             ProtocolEvent::NewInboundSubstream { peer_id, substream } => {
-                self.spawn_inbound_handler(peer_id, substream);
+                // TODO: we need to limit the number of sessions we handle - switch to using RPC?
```

Comment:
I think this is fine as the number of futures that can be active to handle inbound tasks is defined by `max_concurrent_inbound_tasks`. Also, during stress test conditions, out of 4,050 mempool protocol responders that were started in `start_inbound_handler(&mut self, peer_id: PeerId, substream: Substream)`, only 2 failed due to the peer sending an emoty request. 
```suggestion
```

Reply:
Thanks, I added a limit to inbound handlers and forgot to delete this comment